STORAGE: Update API spec to v1.0.0

boonwj opened the pull request:
Changes
- Hologram metadata is nested within `metadata` property in Hologram
  Schema
- Patient metadata is nested within `metadata` property in Patient
  Schema
- `/patients` and `/patients/{pid}` results will contain hologram
  `author` information

New Additions
- Add `FROM_DEPTHVISOR_RECORDING` to `creationMode` enum
- Query field `creationmode` for filtering results based on
  `creationMode` of holograms on `/patients` and `/holograms` endpoints
- Error message returned on errors

Pending major changes, this will end with a commit in the PR to update the spec version to v1.0.0

boonwj committed:
STORAGE: Update API spec to v0.2.0

Changes
- Hologram metadata is nested within `metadata` property in Hologram
  Schema
- Patient metadata is nested within `metadata` property in Patient
  Schema
- `/patients` and `/patients/{pid}` results will contain hologram
  `author` information

New Additions
- Add `FROM_DEPTHVISOR_RECORDING` to `creationMode` enum
- Query field `creationmode` for filtering results based on
  `creationMode` of holograms on `/patients` and `/holograms` endpoints
- Error message returned on errors

Pending major changes, this will soon be updated to v1.0.0

boonwj committed:
STORAGE: Remove feedback in spec

Feedback will be tracked in OneNote instead

nielsboecker reviewed (approved):
Looks great!
Some notes:
1. In `POST /holograms` metadata, do you currently expect any object (that we yet have to define) or the metadata that holograms have when you consume them? I thought that some data might be added by the Accessor implicitly (like creation time, file size...). But maybe it would be easiest to populate all fields on the UI / HoloPipelines before it gets handed to the Storage, because some fields definitely have to be filled, so why not all.
1. Author description struck me funny (`Who and/or what authored the Hologram`)
1. Just to reiterate: We decided _not_ to have `GET /holograms?pid=<pids>`, and the only way to get holograms for given patients is to grab the whole patient resources which will have nested holograms. Correct? Not sure if we should explicitly point this out somewhere or if it is obvious.
1. So when you `GET /holograms*` it always has the nested patient info, when you get the holograms as nested parts of `GET /patients*` the `hologram.subject` is omitted. Correct? Not sure if we should explicitly point this out somewhere or if it is obvious.
1. One obvious shortcoming is the lack of `PUT /holograms`, but that is in correspondance with our requirements. Was thinking I will point this one out in the mail to the other teams, and mention that this may be a later addition to the API if we find the time.

boonwj committed:
STORAGE: Change openapi spec to v0.3.0

Fundamental change to how the schema works, individual patient, hologram
and author resource endpoints now only return their information.

Onus is down to the user to requests additional information via the
other endpoints.

nielsboecker commented on HoloStorageAccessor/api_spec/openapi.yaml:
```diff
@@ -63,24 +66,31 @@ paths:
+
+        Additional filtering for CreationMode can be done with `&creationmode=...`
+
+        Using neither `?pid` nor `?hid` will result in an empty result set.
```
Could mention that using both at once is also not allowed

boonwj replied:
This has been added in a note above.

nielsboecker commented on HoloStorageAccessor/api_spec/openapi.yaml:
```diff
@@ -163,14 +181,92 @@ paths:
+    get:
+      summary: Mass query for authors metadata in HoloStorage
+      description: |-
+        Obtain patient information stored within HoloStorage.
```
author information

nielsboecker commented on HoloStorageAccessor/api_spec/CHANGELOG.md:
```diff
@@ -1,6 +1,35 @@
+### Added
+- `/authors` endpoint to query for author information, available operations are similar to `/patients`
+    - PUT: /authors
+    - GET: /authors?aid=
```
`/authors?aid=id1,id2,id3,...,idx` would be more explicit

boonwj replied:
This has been updated

nielsboecker commented on HoloStorageAccessor/api_spec/openapi.yaml:
```diff
@@ -163,14 +181,92 @@ paths:
+
+      responses:
+        200:
+          description: Successful pulling of patients metadata
```
authors

nielsboecker commented on HoloStorageAccessor/api_spec/openapi.yaml:
```diff
@@ -89,8 +99,8 @@ paths:
+        500:
+          $ref: "#/components/responses/InternalError"
 
     post:
```
Just noticed: A successfull `POST` should return `202 Created`

boonwj replied:
Do you mean 201?

boonwj replied:
done

boonwj committed:
STORAGE: Change to /hologram endpoints

- Change upload data format to POST endpoint
- Change author and patient field names in hologram metadata
- Change successful post status code from 200 to 201

Version update from v0.3.0 to v0.3.1

boonwj committed:
STORAGE: Change directory name of api folder

boonwj committed:
STORAGE: Change hologram data fields and response

Changes to hologram and hologramUpload datatype
- Add bodySite
- Add dateOfImaging
- Add creationDescription
- Remove pipelineId (replaced by creationDescription)

Change POST response to /holograms to hologram datatype

boonwj committed:
STORAGE: record missed changes and update example

- creationDate change missed in changelog
- Update example of creationDescription

boonwj committed:
STORAGE: API version 1.0.0

Bump version number to finalise api version
